### PR TITLE
ACS-8761 Noop Auth Module for unit testing ergonomics

### PR DESCRIPTION
**Pull request by DenysVuika**
**Please check if the PR fulfills these requirements**

> - [x] The commit message follows our [guidelines](https://github.com/Alfresco/alfresco-ng2-components/wiki/Commit-format)
> - [ ] Tests for the changes have been added (for bug fixes / features)
> - [ ] Docs have been added / updated (for bug fixes / features)

**What kind of change does this PR introduce?** (check one with "x")

> - [ ] Bugfix
> - [x] Feature
> - [ ] Code style update (formatting, local variables)
> - [ ] Refactoring (no functional changes, no api changes)
> - [ ] Build related changes
> - [ ] Documentation
> - [ ] Other... Please describe:

**What is the current behaviour?** (You can also link to an open issue here)

**What is the new behaviour?**

Enable testbed reset after each test

```
getTestBed().initTestEnvironment(BrowserDynamicTestingModule, platformBrowserDynamicTesting(), {
    teardown: { destroyAfterEach: true }
});
```

Introduce a `NoopAuthModule` that provides all the necessary testing setup to avoid using:

- ADF AuthModule
- HttpClientTestingModule
- RouterTestingModule
- AppConfigServiceMock (AppConfigService)
- CookieServiceMock (CookieService)
- (new) NoopRedirectAuthService (RedirectAuthService)
- StoragePrefixFactory

**Does this PR introduce a breaking change?** (check one with "x")

> - [ ] Yes
> - [x] No

If this PR contains a breaking change, please describe the impact and migration path for existing applications: ...

**Other information**:
https://hyland.atlassian.net/browse/ACS-8761

### CODE REVIEW

File: lib/process-services/src/lib/people/components/people/people.component.spec.ts [hunk 1]
```diff
@@ -99,33 +96,84 @@ describe('PeopleComponent', () => {
...
+
+            peopleComponent.involveUser(fakeUser);
+
+            fixture.detectChanges();
+
+            expect(peopleComponent.people.length).toBe(2);
+            expect(peopleComponent.error.emit).toHaveBeenCalledWith('Impossible to involve user with task');
+        });
+
+        // it('should involve people', () => {
```
Comment (g-jaskowski) [inline lib/process-services/src/lib/people/components/people/people.component.spec.ts:149]: Missed commented test

**Review by g-jaskowski — APPROVED** [review]

**Review by tamaragruszka — APPROVED** [review]

File: lib/content-services/src/lib/content-metadata/components/content-metadata-card/content-metadata-card.component.spec.ts [hunk 2]
```diff
@@ -20,22 +20,17 @@ import { By } from '@angular/platform-browser';
...
-import { versionCompatibilityFactory } from '../../../version-compatibility/version-compatibility-factory';
-import { VersionCompatibilityService } from '../../../version-compatibility';
 import { MatDialogModule } from '@angular/material/dialog';
 import { MatSnackBarModule } from '@angular/material/snack-bar';
 import { CategoryService } from '../../../category';
 import { TagService } from '../../../tag';
 import { PropertyDescriptorsService } from '../../public-api';
-import { AlfrescoApiService } from '../../../services/alfresco-api.service';
-import { AlfrescoApiServiceMock } from '../../../mock/alfresco-api.service.mock';
+import { ContentTestingModule } from '@alfresco/adf-content-services';
```
Comment (swapnil-verma-gl) [inline lib/content-services/src/lib/content-metadata/components/content-metadata-card/content-metadata-card.component.spec.ts:33]: Is this import correct? Aren't we already inside content-services module?

Comment (mauriziovitale) [reply lib/content-services/src/lib/content-metadata/components/content-metadata-card/content-metadata-card.component.spec.ts:33]: definitely a mistake

File: lib/content-services/src/lib/content-metadata/components/content-metadata/content-metadata.component.spec.ts [hunk 3]
```diff
@@ -45,6 +36,7 @@ import { TagService } from '../../../tag/services/tag.service';
 import { CategoryService } from '../../../category/services/category.service';
 import { TagsCreatorComponent, TagsCreatorMode } from '../../../tag';
 import { CategoriesManagementComponent, CategoriesManagementMode } from '../../../category';
+import { ContentTestingModule } from '@alfresco/adf-content-services';
```
Comment (mauriziovitale) [inline lib/content-services/src/lib/content-metadata/components/content-metadata/content-metadata.component.spec.ts:39]: same

File: lib/content-services/src/lib/content-metadata/components/content-metadata-card/content-metadata-card.component.spec.ts (hunk 2 above)
Comment (DenysVuika) [reply lib/content-services/src/lib/content-metadata/components/content-metadata-card/content-metadata-card.component.spec.ts:33]: IntelliJ being drunk again, will fix :)

File: lib/content-services/src/lib/content-metadata/components/content-metadata/content-metadata.component.spec.ts (hunk 3 above)
Comment (DenysVuika) [reply lib/content-services/src/lib/content-metadata/components/content-metadata/content-metadata.component.spec.ts]: fixed, thanks

**Comment by swapnil-verma-gl** [comment]
The incorrect import issue is present for pretty much all import of NoopTranslateModule and NoopAuthModule inside ADF-Core. Additionally, there are a couple of more occurrences of the issue in some component imports, that I've tagged individually. Many of these are pre-existing imports though, just FYI

File: lib/content-services/src/lib/viewer/components/alfresco-viewer.component.spec.ts [hunk 4]
```diff
@@ -21,29 +21,36 @@ import { ComponentFixture, fakeAsync, TestBed, tick } from '@angular/core/testin
 import { MatDialog, MatDialogModule } from '@angular/material/dialog';
 import { AppExtensionService, ViewerExtensionRef } from '@alfresco/adf-extensions';
 import { ContentInfo, Node, NodeEntry, VersionEntry } from '@alfresco/js-api';
-import { AlfrescoViewerComponent, ContentService, NodeActionsService, RenditionService } from '@alfresco/adf-content-services';
+import { AlfrescoViewerComponent, ContentService, ContentTestingModule, NodeActionsService, RenditionService } from '@alfresco/adf-content-services';
```
Comment (swapnil-verma-gl) [inline lib/content-services/src/lib/viewer/components/alfresco-viewer.component.spec.ts:24]: Incorrect import

File: lib/core/src/lib/dialogs/unsaved-changes-dialog/unsaved-changes.guard.spec.ts [hunk 5]
```diff
@@ -16,16 +16,16 @@
  */
 
 import { TestBed } from '@angular/core/testing';
-import { CoreTestingModule, UnsavedChangesDialogComponent, UnsavedChangesGuard } from '@alfresco/adf-core';
-import { MatDialog, MatDialogRef } from '@angular/material/dialog';
+import { UnsavedChangesDialogComponent, UnsavedChangesGuard } from '@alfresco/adf-core';
```
Comment (swapnil-verma-gl) [inline lib/core/src/lib/dialogs/unsaved-changes-dialog/unsaved-changes.guard.spec.ts:19]: Incorrect import

File: lib/core/src/lib/templates/empty-content/empty-content.component.spec.ts [hunk 6]
```diff
@@ -20,10 +20,12 @@ import { ComponentFixture, TestBed } from '@angular/core/testing';
 import { By } from '@angular/platform-browser';
 import { TranslateService } from '@ngx-translate/core';
 import { of } from 'rxjs';
-import { CoreTestingModule } from '../../testing/core.testing.module';
+import { EmptyContentComponent, NoopTranslateModule } from '@alfresco/adf-core';
```
Comment (swapnil-verma-gl) [inline lib/core/src/lib/templates/empty-content/empty-content.component.spec.ts:23]: Incorrect import

File: lib/core/src/lib/testing/noop-auth.module.ts [hunk 7]
```diff
@@ -0,0 +1,53 @@
...
+import { APP_INITIALIZER, Injectable, NgModule } from '@angular/core';
+import { AuthModule, RedirectAuthService } from '../auth';
+import { HttpClientTestingModule } from '@angular/common/http/testing';
+import { RouterTestingModule } from '@angular/router/testing';
+import { AppConfigService, StoragePrefixFactory } from '../app-config';
+import { AppConfigServiceMock, CookieService, StorageService } from '../common';
+import { CookieServiceMock } from '../mock';
+import { EMPTY, of } from 'rxjs';
+import { loadAppConfig } from '../app-config/app-config.loader';
+import { AdfHttpClient } from '@alfresco/adf-core/api';
```
Comment (swapnil-verma-gl) [inline lib/core/src/lib/testing/noop-auth.module.ts:27]: Possibly incorrect import

File: lib/core/src/lib/viewer/components/viewer-render/viewer-render.component.spec.ts [hunk 8]
```diff
@@ -20,16 +20,18 @@ import { Location } from '@angular/common';
...
-import { MatButtonModule } from '@angular/material/button';
-import { MatDialog } from '@angular/material/dialog';
-import { MatIconModule } from '@angular/material/icon';
+import { MatDialog, MatDialogModule } from '@angular/material/dialog';
 import { By } from '@angular/platform-browser';
-import { CoreTestingModule } from '../../../testing';
+import { NoopTranslateModule } from '../../../testing';
 import { RenderingQueueServices } from '../../services/rendering-queue.services';
 import { ViewerRenderComponent } from './viewer-render.component';
+import { ViewerExtensionDirective } from '@alfresco/adf-core';
```
Comment (swapnil-verma-gl) [inline lib/core/src/lib/viewer/components/viewer-render/viewer-render.component.spec.ts:28]: Incorrect import

File: lib/content-services/src/lib/viewer/components/alfresco-viewer.component.spec.ts (hunk 4 above)
Comment (DenysVuika) [reply lib/content-services/src/lib/viewer/components/alfresco-viewer.component.spec.ts:24]: thanks, fixed

File: lib/core/src/lib/dialogs/unsaved-changes-dialog/unsaved-changes.guard.spec.ts (hunk 5 above)
Comment (DenysVuika) [reply lib/core/src/lib/dialogs/unsaved-changes-dialog/unsaved-changes.guard.spec.ts:19]: fixed, thanks

File: lib/core/src/lib/testing/noop-auth.module.ts (hunk 7 above)
Comment (DenysVuika) [reply lib/core/src/lib/testing/noop-auth.module.ts:27]: this is a secondary package entry

**Pull request merged**